press/api/client.py

```diff
@@ -152,6 +182,34 @@ def get(doctype, name):
 		if doc.team != frappe.local.team().name:
 			raise_not_permitted()
 
+	if doctype in ["Site", "Release Group", "Server"] and not frappe.local.system_user():
+
+		# check if user has a role and the role has any allowed items
+		if roles := get_permission_roles():
+			field = doctype.lower().replace(" ", "_")
+
+			is_permitted = frappe.get_value(
+				"Press Role Permission",
+				filters={
+					"role": ["in", roles],
+					field: name,
+				},
+			)
+			if not is_permitted:
+				raise_not_permitted()
+
+	elif doctype == "Marketplace App" and not frappe.local.system_user():
+		if roles := get_permission_roles():
+			if not any(
+				frappe.get_all(
+					"Press Role",
+					fields=["enable_apps"],
+					filters={"name": ("in", roles)},
+					pluck="enable_apps",
+				)
+			):
+				raise_not_permitted()
+
```

Comment:
this block of code is similar to the one in get_list with minor changes, can you figure out a way to commonify?